turbo/consumers.py

```diff
@@ -47,16 +40,25 @@ def notify(self, event, *args, **kwargs):
             })
 
     def receive_json(self, content, **kwargs):
-        model_label = content.get("model")
-        pk = content.get("pk")
+        signer = Signer()
         request_id = content.get("request_id")
+        message_type = content.get("type")
+        if message_type == "subscribe":
+            try:
+                channel_name = signer.unsign(content.get("signed_channel_name", ""))
+            except BadSignature:
+                print("Signature has been tampered with!")
+                return
 
-        channel_name = make_channel_name(model_label, pk)
+            self.requests.setdefault(channel_name, []).append(request_id)
+            self.groups.append(channel_name)
+            async_to_sync(self.channel_layer.group_add)(channel_name, self.channel_name)
+        elif message_type == "unsubscribe":
+            try:
+                channel_name = [channel_name for channel_name, requests in self.requests.items() if request_id in requests][0]
+            except IndexError:
+                return  # No subscription for a given request ID exists.
```

Comment:
I might be interested in this failing loudly as well. Is there a reason for this to ever happen unless there's a programming error?

Reply:
Ditto! No reason this will fail under normal circumstances.